ion/reader.go

```diff
@@ -155,14 +155,25 @@ type Reader interface {
 
 	// StringValue returns the current value as a string (if that makes sense). It returns
 	// an error if the current value is not an Ion symbol or an Ion string.
-	StringValue() (string, error)
+	StringValue() (*string, error)
```

Comment:
This really simplifies [the case from my earlier comment](https://github.com/amzn/ion-go/pull/149#discussion_r481272560); if the string or symbol was `null`, this function will return a `nil` that can be passed around. Great.

However, we now have a larger consistency question for the `Reader` interface. Should all values return `*type` so they return a `nil` if the Ion value is `null`? At the moment most of them return an `err` with the expectation that you will have handled `null` separately through a combination of `IsNull()` and `Type()`. (This isn't documented yet, by the way. See [`IntValue()` for an example](https://github.com/amzn/ion-go/blob/1f421dfa4f274ec841989150ea325434170aac0f/ion/reader.go#L125-L128) -- the doc comment doesn't explain what to expect if the value is `null`.)

Having two layers of checking to do after calling each `*Value()` function (Is it an `err`? Is it `nil`?) feels pretty heavy to me, which is why I think we were generating a `UsageError` for `null` values originally.

This is a larger discussion, but it's something I think we need to get right before 1.0. I'm ok with opening an issue to track this so it doesn't block the PR.

Reply:
Created ticket: https://github.com/amzn/ion-go/issues/152